libs/nx-remotecache-s3/src/lib/s3-client.ts

```diff
@@ -29,6 +56,11 @@ export const buildS3Client = (
     credentials: provider,
     forcePathStyle:
       getEnv(ENV_FORCE_PATH_STYLE) === 'true' || options.forcePathStyle,
+    ...getProxyConfig(
+      getEnv(ENV_ENDPOINT) ??
```

Comment:
OK I started to do this, which lead me down a rabbit hole to remember exactly why I added it.  I'm going to add more into the readme on the endpoint and no_proxy usage, but the short of it is we don't want to force the s3 client into a specific endpoint unless the user actually configures one.  So, I'm going to leave this function as is.

Reply:
I agree, it's just that it seems like there is no guarantee that the default endpoint that is hardcoded here (https://amazonaws.com) to determine whether to bypass the proxy is also the actual endpoint that is used in case no endpoint is provided. Ideally there would be a way to retrieve the default endpoint from the sdk, but I'm not sure it's possible?

Reply:
That is exactly the issue; unfortunately, aws sdk libraries don't provide any api to get the endpoint that the client will use.   I'm adding that to the readme and an explanation for the user on how to fix no_proxy issues if in fact they need to bypass the proxy but are seeing problems.